Tests/SwiftParserTest/TrailingCommaTests.swift

```diff
@@ -0,0 +1,96 @@
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(ExperimentalLanguageFeatures) import SwiftParser
+import XCTest
+
+final class TrailingCommaTests: ParserTestCase {
+  
+  func testTuple() {
+    assertParse("(1, 2, 3,)", experimentalFeatures: .trailingComma)
+    assertParse(
+      "(1️⃣,)",
+      diagnostics: [DiagnosticSpec(message: "expected value in tuple", fixIts: ["insert value"])],
+      fixedSource: "(<#expression#>,)",
+      experimentalFeatures: .trailingComma
+    )
+  }
+  
+  func testArgumentList() {
+    
+    assertParse(
+      """
+      if true, { value = 1 }
+      expectEqual(value, 1)
+      """,
+      experimentalFeatures: .trailingComma
+    )
+    
+    assertParse("f(1, 2, 3,)", experimentalFeatures: .trailingComma)
+    assertParse(
+      "f(1️⃣,)",
+      diagnostics: [DiagnosticSpec(message: "expected value in function call", fixIts: ["insert value"])],
+      fixedSource: "f(<#expression#>,)",
+      experimentalFeatures: .trailingComma
+    )
+  }
+  
+  func testIfConditions() {
+        
+    assertParse("if true, f { $0 }, { true }(), { value = 2 } else { value = 0 }", experimentalFeatures: .trailingComma)
+    assertParse("if conditionA, { value = 1 }", experimentalFeatures: .trailingComma)
+    assertParse("if conditionA, conditionB, { value = 1 }", experimentalFeatures: .trailingComma)
+    assertParse("if conditionA, {x}(), { value = 1 }", experimentalFeatures: .trailingComma)
+    assertParse("if conditionA, { a in a == 1 }(1), { value = 1 }", experimentalFeatures: .trailingComma)
+    assertParse("if conditionA, conditionB, { value = 1 } else if conditionC, {x}(), { value = 2 }", experimentalFeatures: .trailingComma)
+    
+    assertParse(
+      "if 1️⃣,{}",
+      diagnostics: [DiagnosticSpec(message: "missing condition in 'if' statement")],
+      experimentalFeatures: .trailingComma
+    )
+    
+    assertParse("if conditionA, {}, {}", experimentalFeatures: .trailingComma)
+    
+    assertParse(
+      """
+      value = 0
+      if true, { value = 1 }
+      expectEqual(value, 1)
+      """,
+      experimentalFeatures: .trailingComma
+    )
+    
+  }
+  
+  func testGuardConditions() {
+    assertParse("guard conditionA, else { break }", experimentalFeatures: .trailingComma)
+    assertParse("guard conditionA, conditionB, else { break }", experimentalFeatures: .trailingComma)
+    assertParse("guard conditionA, {x}(), else { break }", experimentalFeatures: .trailingComma)
+    assertParse("guard conditionA, { a in a == 1 }(1), else { break }", experimentalFeatures: .trailingComma)
+  }
+  
+  func testWhileConditions() {
+    assertParse(
+      """
+      var value = 5
+      while value != 0, {
+          value -= 1
+      }
+      expectEqual(value, 0)
+      """,
+      experimentalFeatures: .trailingComma
+    )
+    assertParse("while conditionA, {x}(), { value += 1 }", experimentalFeatures: .trailingComma)
+  }
+  
```

Comment:
Another interesting test case is

```swift
if foo { print("a") }, else: { print("b") } {
}
```

Where `else:` is supposed to be an additional trailing closure, `foo` being defined as follows

```swift
func foo(_: () -> Void, `else`: () -> Void) -> Bool { true }
```

Also worth adding a test case where the trailing closure label is something normal like `bar`.

Reply:
By the way, currently `if foo { print("a") } else: { print("b") } { }` doesn't parse correctly. Is this intended or should be fixed by modifying [atValidTrailingClosure](https://github.com/apple/swift-syntax/blob/25ce3a297818076ceab8054873a6189e8b8b6688/Sources/SwiftParser/Expressions.swift#L1983) when flavor is `stmtCondition`?

Reply:
Define “correctly” ;-). It’s a super weird edge case and I think the current behavior or parsing `else` as the start of an `else` block is the best solution IMO. Parsing `else` as an additional trailing closure makes thinks even more confusing IMO. 

I would mostly like to make sure that we don’t change the current behavior.